pkg/diagnose/check.go

```diff
@@ -110,7 +111,7 @@ func diagnoseChecksInCLIProcess(diagCfg diagnosis.Config, senderManager diagnose
 	secretResolver := secretsimpl.GetInstance()
 
 	// Initializing the aggregator with a flush interval of 0 (to disable the flush goroutines)
-	common.LoadComponents(senderManagerInstance, secretResolver, pkgconfig.Datadog.GetString("confd_path"))
+	common.LoadComponents(senderManagerInstance, secretResolver, workloadmeta.GetGlobalStore(), pkgconfig.Datadog.GetString("confd_path"))
```

Comment:
Do we have plan to remove GetGlobalStore here?

Reply:
Ideally yes, but it would need changes to how diagnose works (removing init method). I am evaluating if this is doable in a reasonable amount of time or not.

Reply:
I guess you're holding a list of things to fix locally, but if it's not the case I think we should add a TODO(components) comment to not forget

Reply:
This is tracked by ASCII-1162 which implies removing `GetGlobalStore`.